xmake/rules/c++/modules/modules_support/msvc.lua

```diff
@@ -127,27 +128,30 @@ function load(target)
     local modulesflag = get_modulesflag(target)
     target:add("cxxflags", modulesflag)
 
-    -- add stdifcdir in case of if the user ask for it
-    local stdifcdirflag = get_stdifcdirflag(target)
-    if stdifcdirflag then
+    -- enable std modules if c++23 by defaults
+    if target:values("c++.msvc.enable_std_import") == nil then
+        local languages = table.join({}, target:get("languages"))
+        local iscpp23 = false
+        for _, language in pairs(languages) do
+            if language:find("c++") or language:find("cxx") then
+                iscpp23 = true
+                for _, version in pairs({"11", "14", "17", "20"}) do
+                    if language:find(version) then
+                        iscpp23 = false
+                        break
+                    end
+                end
+            end
+        end
+        local stdmodulesdir
         local msvc = target:toolchain("msvc")
         if msvc then
             local vcvars = msvc:config("vcvars")
-            if vcvars.VCInstallDir and vcvars.VCToolsVersion then
-                local arch
-                if target:is_arch("x64", "x86_64") then
-                    arch = "x64"
-                elseif target:is_arch("x86", "i386") then
-                    arch = "x86"
-                end
-                if arch then
-                    local stdifcdir = path.join(vcvars.VCInstallDir, "Tools", "MSVC", vcvars.VCToolsVersion, "ifc", arch)
-                    if os.isdir(stdifcdir) then
-                        target:add("cxxflags", {stdifcdirflag, winos.short_path(stdifcdir)}, {force = true, expand = false})
-                    end
-                end
+            if vcvars.VCInstallDir and vcvars.VCToolsVersion and semver.compare(vcvars.VCToolsVersion, "14.35") then
+                stdmodulesdir = path.join(vcvars.VCInstallDir, "Tools", "MSVC", vcvars.VCToolsVersion, "modules")
             end
         end
+        target:set("values", "c++.msvc.enable_std_import", iscpp23 and os.isdir(stdmodulesdir))
```

Comment:
if it's only private data in rule, you should use `target:data_set()` and `target:data()`

Reply:
it's not private, i think some users may need to disable std modules